src/lib.rs

```diff
@@ -442,6 +451,20 @@ impl Mock {
 
         true
     }
+
+    fn body_matches(&self, request: &mut Request) -> bool {
+        if !self.request.body.is_empty() {
+            let mut body = String::new();
+            request.read_to_string(&mut body).unwrap();
+
+            let actual_json: Value = serde_json::from_str(&body).unwrap();
+            let expected_json: Value = serde_json::from_str(&self.request.body).unwrap();
+
+            expected_json == actual_json
```

Comment:
what's the benefit of comparing JSON objects, instead of Strings?
aside from the key order (?!) I can only see a benefit here if you'd implement partial comparison (expected JSON is included in the actual JSON), which would be pretty awesome actually.